module16/test_cases.robot

```diff
@@ -0,0 +1,87 @@
+*** Settings ***
+Library    SeleniumLibrary
+Suite Teardown    Close All Browsers
+
+*** Variables ***
+${user_login}    gryundelb@gmail.com
+${user_password}    Password1
+
+*** Test Cases ***
+Test Login
+    [Tags]    screenshot
+    Open Browser    https://www.demoblaze.com/    Chrome
+    Set Selenium Speed    0.5s
+    Maximize Browser Window
+    Click Login Button
+    Wait Until Element Is Visible    css=#loginusername
+    Wait Until Element Is Visible    css=#loginpassword
+    Log In
+    Wait Until Element Is Visible    css=#logout2
+    ${welcome_message}=    Get Text    css=#nameofuser
+    Should Be Equal As Strings    ${welcome_message}    Welcome ${user_login}
+    Capture Page Screenshot
+
+Test Add Product To Cart
+    [Tags]    screenshot
+    Open Browser    https://www.demoblaze.com/    Chrome
+    Set Selenium Speed    0.5s
+    Maximize Browser Window
+    Click Login Button
+    Wait Until Element Is Visible    css=#loginusername
+    Wait Until Element Is Visible    css=#loginpassword
+    Log In
+    Open Monitors Category
+    ${product_info}=    Get Highest Price Monitor
+    Click Add To Cart Button
+    Click Cart Button
+    Wait Until Element Is Visible    css=.success
+    ${cart_product_info}=    Get Product Info
+    Should Be Equal As Strings    ${product_info['name']}    ${cart_product_info['name']}
+    Should Be Equal As Strings    ${product_info['price']}    ${cart_product_info['price']}
+    Capture Page Screenshot
+
+*** Keywords ***
+Login
+    Input Text    css=#loginusername    ${user_login}
+    Input Text    css=#loginpassword    ${user_password}
+    Click Button    //button[text()='Log in']
+
+Click Login Button
+    Click Element    css=#login2
+
+Open Monitors Category
+    Click Link    //a[text()='Monitors']
+
+Get Highest Price Monitor
```

Comment:
It would be better use python helper modules to keep complicated keywords like this

Reply:
Done. Added custom_keywords.py to Get Highest Price Monitor